tests/acceptance/guid-file/registration-file-detail-test.ts

```diff
@@ -98,7 +126,162 @@ module('Acceptance | guid file | registration files', hooks => {
+            'File affiliated institutions text is properly set.');
+        assert.dom('[data-test-target-institutions]').exists();
+    });
+
+    // Test metadata save and cancel buttons
+    test('save and cancel', async function(this: ThisTestContext, assert) {
+        await visit(`/--file/${this.file.id}`);
+        assert.equal(currentURL(), `/--file/${this.file.guid}`);
+
+        // Verify download
+        if (!this.node._anonymized) {
+            assert.dom('[data-test-download-button]').exists();
+        }
+
+        // Verify edit
+        if (this.file.target.currentUserPermissions.includes(Permission.Write)) {
+            await click('[data-test-edit-metadata-button]');
+            // Verify editable fields are present
+            assert.dom('[data-test-title-field]').exists();
+            assert.dom('[data-test-description-field]').exists();
+            assert.dom('[data-test-select-resource-type]').exists();
+            assert.dom('[data-test-select-resource-language]').exists();
+            assert.dom('[data-test-cancel-editing-metadata-button]').exists();
+            assert.dom('[data-test-save-metadata-button]').exists();
+            await click('[data-test-cancel-editing-metadata-button]');
+            assert.dom('[data-test-edit-metadata-form]').isNotVisible();
+            await click('[data-test-edit-metadata-button]');
+
+            // Screenshot before changes
+            await percySnapshot(assert);
+            // Update Title
+            const editTitleTextArea = document.querySelectorAll('textarea')[0];
+            editTitleTextArea.textContent = 'A test title.';
+            assert.dom('[data-test-title-field]')
+                .containsText('A test title.', 'Metadata title field updates.');
+            // Update Description
+            const editDescriptionTextArea = document.querySelectorAll('textarea')[1];
+            editDescriptionTextArea.textContent = 'A test description.';
+            assert.dom('[data-test-title-field]')
+                .containsText('A test description.', 'Metadata description field updates.');
+            // Update Resource Type
+            const resourceTypeDropdown = document.querySelectorAll('[data-test-power-select-dropdown]')[0];
+            await click(resourceTypeDropdown);
+            const resourceTypeList = document.evaluate(
+                'count(//ul[@role="listbox"]/li)',
+                document,
+                null,
+                XPathResult.ANY_TYPE, null,
+            );
+            const resourceTypeListLength = resourceTypeList.numberValue;            // should be in range 0-tens
+            const resourceOptionRandom = Math.floor(resourceTypeListLength * Math.random() - 1);
+            const resourceOption = `data-option-index="${resourceOptionRandom}"`;
+            const resourceElement = document.querySelectorAll(resourceOption)[0];
+            await click(resourceElement);
```

Comment:
While this is very slick, and you clearly put some effort into the implementation, I would advise against doing this. In general, using random numbers in tests is unhelpful and sometimes very bad. There's no good reason why selecting one option might fail while another would pass, and if it did, it would only happen probabilistically with not enough information to debug why it's failing.

Instead, you should evaluate if there's a reason why it might fail for one option over another and, if there's a good reason why it might, set up tests for both of those options. If there's not, one test with one option should do fine.

The same comment goes for any other places you are randomly selecting an option. Best to just use the `selectChoose` helper from `ember-power-select/test-support` to select the options, as well. It should save you some trouble.

Reply:
Thank you Brian; this test is now updated.